tests/functional/lib/receptorcontrol/receptorcontrol.go

```diff
@@ -158,3 +165,152 @@ func (r *ReceptorControl) Status() (*netceptor.Status, error) {
 
 	return &status, nil
 }
+
+// WorkSubmit begins work on remote node
+func (r *ReceptorControl) WorkSubmit(node, serviceName string) (string, error) {
+	_, err := r.WriteStr(fmt.Sprintf("work submit %s %s\n", node, serviceName))
+	if err != nil {
+		return "", err
+	}
+	r.ReadStr()
+	r.CloseWrite()
+	response, err := r.ReadAndParseJSON()
+	if err != nil {
+		return "", err
+	}
+	unitID := fmt.Sprintf("%v", response["unitid"])
```

Comment:
We might consider returning the whole response here and updating `ReadAndParseJSON` to return `map[string]interface{}`

Reply:
i think we also want to close the whole socket here after we get our response?